cli/azd/pkg/project/service_target_aks.go

```diff
@@ -0,0 +1,458 @@
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed parsing kube config: %w", err)
+	}
+
+	if err := kubeConfigManager.SaveKubeConfig(ctx, clusterName, kubeConfig); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed saving kube config: %w", err)
+	}
+
+	if err := kubeConfigManager.MergeConfigs(ctx, "config", "config", clusterName); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed merging kube configs: %w", err)
+	}
+
+	if _, err := t.kubectl.ConfigUseContext(ctx, clusterName, nil); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed using kube context '%s', %w", clusterName, err)
+	}
+
+	namespace := t.getK8sNamespace()
+	kubeFlags := kubectl.KubeCliFlags{
+		Namespace: namespace,
+		DryRun:    "client",
+		Output:    "yaml",
+	}
+
+	progress <- "Creating k8s namespace"
+	namespaceResult, err := t.kubectl.CreateNamespace(
+		ctx,
+		namespace,
+		&kubectl.KubeCliFlags{DryRun: "client", Output: "yaml"},
+	)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed creating kube namespace: %w", err)
+	}
+
+	_, err = t.kubectl.ApplyPipe(ctx, namespaceResult.Stdout, nil)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed applying kube namespace: %w", err)
+	}
+
+	progress <- "Creating k8s secrets"
+	secretResult, err := t.kubectl.CreateSecretGenericFromLiterals(ctx, "azd", t.env.Environ(), &kubeFlags)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed setting kube secrets: %w", err)
+	}
+
+	_, err = t.kubectl.ApplyPipe(ctx, secretResult.Stdout, nil)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed applying kube secrets: %w", err)
+	}
+
+	imageTag, err := t.generateImageTag()
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed generating image tag: %w", err)
+	}
+
+	fullTag := fmt.Sprintf(
+		"%s/%s",
+		loginServer,
+		imageTag,
+	)
+
+	// Tag image.
+	log.Printf("tagging image %s as %s", path, fullTag)
+	progress <- "Tagging image"
+	if err := t.docker.Tag(ctx, t.config.Path(), path, fullTag); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed tagging image: %w", err)
+	}
+
+	log.Printf("pushing %s to registry", fullTag)
+
+	// Push image.
+	progress <- "Pushing container image"
+	if err := t.docker.Push(ctx, t.config.Path(), fullTag); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed pushing image: %w", err)
+	}
+
+	// Save the name of the image we pushed into the environment with a well known key.
+	t.env.SetServiceProperty(t.config.Name, "IMAGE_NAME", fullTag)
+
+	if err := t.env.Save(); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("saving image name to environment: %w", err)
+	}
+
+	progress <- "Applying k8s manifests"
+	t.kubectl.SetEnv(t.env.Values)
+	deploymentPath := t.config.K8s.DeploymentPath
+	if deploymentPath == "" {
+		deploymentPath = defaultDeploymentPath
+	}
+
+	err = t.kubectl.ApplyFiles(
```

Comment:
Let's remember to explicitly document that the order in which these manifests are applied (alphabetically, it seems since os.ReadDir sorts its results).

Reply:
Order shouldn't matter as k8s handles this and the deployed services, etc follow an eventual consistency model.